test/Npgsql.Tests/Types/JsonPathTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Threading.Tasks;
+using NpgsqlTypes;
+using NUnit.Framework;
+using static Npgsql.Tests.TestUtil;
+
+namespace Npgsql.Tests.Types
+{
+    public class JsonPathTests : MultiplexingTestBase
+    {
+        public JsonPathTests(MultiplexingMode multiplexingMode)
+            : base(multiplexingMode) { }
+
+        static readonly object[] ReadWriteCases = new[]
+        {
+            new object[] { "'$'::text", "$" },
+            new object[] { "'$\"varname\"'::text", "$\"varname\"" },
+        };
+
+        [Test]
+        [TestCaseSource(nameof(ReadWriteCases))]
+        public async Task Read(string query, string expected)
+        {
+            using var conn = await OpenConnectionAsync();
+            MinimumPgVersion(conn, "12.0", "The jsonpath type was introduced in PostgreSQL 12");
+
+            using var cmd = new NpgsqlCommand("SELECT " + query, conn);
+            using var rdr = await cmd.ExecuteReaderAsync();
+
+            rdr.Read();
+            Assert.That(rdr.GetFieldValue<string>(0), Is.EqualTo(expected));
+            Assert.That(rdr.GetTextReader(0).ReadToEnd(), Is.EqualTo(expected));
+        }
+
+        [Test]
+        [TestCaseSource(nameof(ReadWriteCases))]
+        public async Task Write(string query, string expected)
+        {
+            using var conn = await OpenConnectionAsync();
+            MinimumPgVersion(conn, "12.0", "The jsonpath type was introduced in PostgreSQL 12");
+
+            using var cmd = new NpgsqlCommand("SELECT @p, @p::text = " + query, conn) { Parameters = { new NpgsqlParameter("p", NpgsqlDbType.JsonPath) { Value = expected } } };
+            using var rdr = await cmd.ExecuteReaderAsync();
+
+            rdr.Read();
+            Assert.That(rdr.GetFieldValue<string>(0), Is.EqualTo(expected));
+            Assert.That(rdr.GetFieldValue<bool>(1));
```

Comment:
nit: can also write Assert.True which I find slightly more clear, but really not important.

Reply:
This test verifies that the server correctly parsed the parameter and it has the same values the unlined into the query. Otherwise, reading the passed parameter may hide a possible bug because of a symmetric in `Write`.

Reply:
I agree it's a good idea to split reading and writing tests (and we don't do that with many types).

But specifically here, isn't it enough to just do `SELECT @p::text` and assert on the returned text representation? That would be a 100% reliable test of the sending, no?

Reply:
In that case we will rely on the behavior of `TextHandler` which could have a bug too.

Reply:
I think it's ok to assume that TextHandler works in this test, no? After all your relying on BoolHandler not having bugs in the exact same way

Reply:
Fixed, it doesn't rely on other handlers any more.